### ISSUE TITLE
What's the pipeline equivalent of the post-build artifactory option

### ISSUE THREAD

**Issue by pauldambra** [jfrog/jenkins-artifactory-plugin#211]
https://stackoverflow.com/questions/58809425/convert-a-jenkins-artifactory-post-build-step-into-a-pipeline-stage

I have the above StackOverflow question. I'm converting a jenkins job which has a build action that runs maven and a separate post-build action that generates build info and publishes maven artefacts. What's the pipeline equivalent?

**Comment by reenberg** [comment]
The jfrog documentation on [Working with pipeline jobs in jenkins](https://www.jfrog.com/confluence/display/RTF/Working+With+Pipeline+Jobs+in+Jenkins?src=sidebar)  contains a dedicated page about [Declarative pipeline syntax](https://www.jfrog.com/confluence/display/RTF/Declarative+Pipeline+Syntax) (and scripted for that matter).
I would suggest that you take a look at that, as it covers what you seem to need, and then perhaps ask again if that source of information isn't enough. 

You most likely don't want to use an upload spec, but configure the build to both deploy your artifact and publish your build information, for example using an `rtMavenResolver` and `rtMavenDeployer` inside an `rtMavenRun`. 

With the upload step you have to manually specify where you want the artifacts to be (the path, as I remember) , thus why you experience them being put at the root.

**Comment by pauldambra** [comment]
Yep. I've read that but it isn't the same. The post build action doesn't
build the software whereas the pipeline stuff wraps the build.

I'd like to keep them separate.

Having built the software separately I'd like to upload it to artifactory.

That documentation and the examples github repo are so far removed from
what I want that I'm assuming it's not possible but checking here.

My build step installs to the build agent's m2 repository. Can I read from
there and copy to artifactory?

**Comment by eyalbe4** [comment]
@pauldambra,
If you'd like the maven artifacts to be deployed at later stage (separate from the actual build), this is supported by the scripted pipeline syntax (not supported currently by the declarative syntax). Read [here](https://www.jfrog.com/confluence/display/RTF/Scripted+Pipeline+Syntax#ScriptedPipelineSyntax-MavenBuildswithArtifactory).
